controllers/memberoperatorconfig/cache.go

```diff
@@ -28,59 +28,61 @@ func (c *cache) set(config *toolchainv1alpha1.MemberOperatorConfig, secrets map[
 	c.Lock()
 	defer c.Unlock()
 	c.config = config.DeepCopy()
-	c.secrets = copyOf(secrets)
+	c.secrets = common.CopyOf(secrets)
 }
 
 func (c *cache) get() (*toolchainv1alpha1.MemberOperatorConfig, map[string]map[string]string) {
 	c.RLock()
 	defer c.RUnlock()
-	return c.config.DeepCopy(), copyOf(c.secrets)
+	return c.config.DeepCopy(), common.CopyOf(c.secrets)
 }
 
 func updateConfig(config *toolchainv1alpha1.MemberOperatorConfig, secrets map[string]map[string]string) {
 	configCache.set(config, secrets)
 }
 
-func loadLatest(cl client.Client) error {
+func loadLatest(cl client.Client) (Configuration, error) {
 	namespace, err := common.GetWatchNamespace()
 	if err != nil {
-		return errs.Wrap(err, "Failed to get watch namespace")
+		return Configuration{m: &toolchainv1alpha1.MemberOperatorConfigSpec{}}, errs.Wrap(err, "Failed to get watch namespace")
 	}
 
 	config := &toolchainv1alpha1.MemberOperatorConfig{}
 	if err := cl.Get(context.TODO(), types.NamespacedName{Namespace: namespace, Name: "config"}, config); err != nil {
 		if apierrors.IsNotFound(err) {
 			cacheLog.Info("MemberOperatorConfig resource with the name 'config' wasn't found, default configuration will be used", "namespace", namespace)
-			return nil
+			return Configuration{m: &toolchainv1alpha1.MemberOperatorConfigSpec{}}, nil
```

Comment:
why don't we call `getConfigOrDefault()` here as well?
```suggestion
			return getConfigOrDefault(), nil
```
so if the config is (accidentally) removed, we could still used the cached one

Reply:
That's a good point. It's more likely that we accidentally remove the config than remove it and want the default configuration to take over so I'll make that change.

Reply:
Done. Updated the code as well as the message and the unit tests. b6313c747449961f0ab297c81f3cd687a35bd6b4 and ef6402100b4f3380cf0fa5ab7694d672f8f031ee